roles/hosted_cluster/tasks/main.yml

```diff
@@ -78,26 +78,82 @@
         infrastructureAvailabilityPolicy: HighlyAvailable
         olmCatalogPlacement: management
 
-- name: Manage NodePool resource
+- name: Manage NodePool resources
   kubernetes.core.k8s:
     state: "{{ hosted_cluster_state }}"
     definition:
       apiVersion: hypershift.openshift.io/v1beta1
       kind: NodePool
       metadata:
-        name: "nodepool-{{ hosted_cluster_name }}"
+        name: "nodepool-{{ hosted_cluster_name }}-{{ item.resource_class }}"
         namespace: "{{ hosted_cluster_namespace }}"
         labels: "{{ hosted_cluster_default_cluster_order_label }}"
       spec:
         clusterName: "{{ hosted_cluster_name }}"
-        replicas: "{{ hosted_cluster_node_count | int }}"
+        replicas: "{{ item.number_of_nodes | int }}"
         management:
           autoRepair: false
           upgradeType: InPlace
         platform:
           type: Agent
           agent:
             agentLabelSelector:
-              matchLabels: "{{ hosted_cluster_default_cluster_order_label }}"
+              matchLabels: >
+                "{{ {esi_agent_resource_class_label: item.resource_class} | combine(hosted_cluster_default_cluster_order_label) }}"
         release:
           image: "{{ ocp_release_image }}"
+  loop: "{{ hosted_cluster_node_requests }}"
+  loop_control:
+    label: "Manage NodePool nodepool-{{ hosted_cluster_name }}-{{ item.resource_class }}"
+
+- name: Create cluster network
+  ansible.builtin.include_role:
+    name: massopencloud.esi.network
+  vars:
+    network_suffix: "{{ hosted_cluster_name }}"
+    network_state: "{{ hosted_cluster_state }}"
+  when: hosted_cluster_state == "present"
```

Comment:
Why does only this one task of a `when` condition on `hosted_cluster_state`?

Reply:
if create (`hosted_cluster_state == "present"`) a hosted cluster, we want to create a network before any agent are attached to it, if we delete (`hosted_cluster_state == "absent"`) a hosted cluster I guess we want to detach all the agents from it before we delete it (the deletion task is at the end of the file).